src/tab.group.ts

```diff
@@ -274,18 +271,30 @@ export default class TabGroup extends LitElement {
     }
   }
 
+  #onDefaultSlotChange() {
+    this.#setAriaAttributes();
+  }
+
   #onNavSlotChange() {
-    this.#setUpTabs();
-    this.#updateTabsPanelsAndTabBar();
+    this.#setAriaAttributes();
+    this.#updateSelectedTabAndPanel();
+    this.#updateSelectedTabIndicator();
     this.#setOverflowButtonsState();
   }
 
-  #onOverflowEndButtonClick() {
-    this.#scrollTabsList('right');
-  }
+  #onOverflowButtonClick(button: 'start' | 'end') {
```

Comment:
Simplified this a bit because it'll likely remain in some form after Tab Group is rewritten.